gnd/src/main/java/com/google/android/gnd/persistence/local/room/RoomLocalDataStore.java

```diff
@@ -436,4 +438,19 @@ public Completable insertOrUpdateTile(Tile tile) {
   public Maybe<Tile> getTile(String tileId) {
     return tileDao.findById(tileId).map(TileEntity::toTile).subscribeOn(schedulers.io());
   }
+
+  @Override
+  public Single<ImmutableList<Tile>> getPendingTiles() {
+    // TODO: Only retrieve tiles for a given area.
+    // This currently retrieves all pending tiles in the DB.
+    return tileDao
+        .findAll()
```

Comment:
Rather than using findAll and filtering, can we define a method on the DAO that only returns tiles with state == PENDING?

Reply:
The DAO Query strings have to be constant, which unfortunately means we can't exactly add a Pending tiles method to the DAO directly without using an int literal, which is too brittle for my taste.

However, I have switched to using a `findByState` method instead, which is simpler!

Reply:
see commit #2b867a8